### PR TITLE
add `ruff` and remove redundant pre-commits

### PR DESCRIPTION
**Pull request by giovp**
replace `flake8` and a bunch of other pre-commit with https://github.com/charliermarsh/ruff, a faster alternative.

If accepted/in scopre, merge only after #269

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -233,3 +233,84 @@ legacy_tox_ini = """
...
+    "docs/*" = ["E", "F", "W", "I", "D101", "D102"]
+    "src/ott/types.py" = ["D102"]
+[tool.ruff.pydocstyle]
+convention = "google"
+[tool.ruff.pyupgrade]
+# Preserve types, even if a file imports `from __future__ import annotations`.
+keep-runtime-typing = true
+[tool.ruff.flake8-tidy-imports]
+# Disallow all relative imports.
+ban-relative-imports = "all"
```
Comment (michalk8) [inline pyproject.toml:316]: I think the value is wrong, see [here](https://github.com/adamchainz/flake8-tidy-imports#ban-relative-imports).
But I also think we wanna keep relative imports, given we have only 80 columns, pinging @marcocuturi .

**Review by michalk8 — CHANGES_REQUESTED** [review]

File: pyproject.toml (hunk 1 above)
Comment (michalk8) [reply pyproject.toml:316]: Ok, you can add this, but please check the value is correct - in `__init__.py`, relative imports are still present.

Comment (giovp) [reply pyproject.toml:316]: ok shall I ban all or just parents?

Comment (giovp) [reply pyproject.toml]: I added `parents` option, so init are not touched

File: pyproject.toml [hunk 2]
```diff
@@ -233,3 +233,84 @@ legacy_tox_ini = """
...
+    # whitespace before : -> black does not adhere to PEP8
+    # "E203",
+    # whitespace before : -> black does not adhere to PEP8
+    # "E203",
+    # missing whitespace after ,', ';', or ':' -> black does not adhere to PEP8
+    # "E231",
+    # continuation line over-indented for hanging indent -> black does not adhere to PEP8
+    # "E126",
+    # too many leading '# ' for block comment -> Scanpy allows them for comments into sections
+    # "E262",
```
[the remark this answers is not in the document]
Comment (giovp) [reply pyproject.toml]: good point, removed

File: pyproject.toml (hunk 1 above)
Comment (michalk8) [reply pyproject.toml]: Thanks!

**Comment by michalk8** [comment]
> now there are `"E","F","W","I",` with autofix, which would introduce way too many changes in `docs` and `tests` and maybe it should be done later on (more stable repo, not today), so I explicitly removed them.

Perfect, can you create another issue after this one is merged for this?
LGTM, merging this since the linter passing, thanks @giovp !

**Pull request merged**